evaluation/EDA/run_infer.py

```diff
@@ -118,7 +118,7 @@ def process_instance(
     instruction += AGENT_CLS_TO_INST_SUFFIX[metadata.agent_class]
 
     # Here's how you can run the agent (similar to the `main` function) and get the final task state
-    runtime = create_runtime(config, sid=instance['text'].strip())
+    runtime = create_runtime(config)
```

Comment:
I think if we just remove it here, it will still be generated automatically and set to https://github.com/All-Hands-AI/OpenHands/blob/ea883d4d1863513de6990c7e80b508059d1a2ce7/openhands/core/main.py#L214

It's not clear to me why that would work? What value do we need now?

Reply:
Basically we are getting errors on the backend for:

```
[start-runtime] Failed to create deployment, service, or ingress: (422)
Reason: Unprocessable Entity
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Deployment.apps \"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9\" is invalid: [metadata.name: Invalid value: \"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.template.spec.containers[0].name: Invalid value: \"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9\": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')]","reason":"Invalid","details":{"name":"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9","group":"apps","kind":"Deployment","causes":[{"reason":"FieldValueInvalid","message":"Invalid value: \"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')","field":"metadata.name"},{"reason":"FieldValueInvalid","message":"Invalid value: \"runtime-d8994ccd-c1bc-4c77-85b8-6647edb13fc7_ec69396697b4e4c9\": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')","field":"spec.template.spec.containers[0].name"}]},"code":422}
```

Before this PR, if you just pass in instance_id, it is very likely that those ids will violate these constraints and RemoteRuntime `/start` would raise 500 error.

And I actually just modified the generate_sid to produce sid delimited by `-` instead of `_` to avoid this error.

Reply:
whoops I broke this by passing the `sid` into the remote runtime

Reply:
Ah, thank you. I'm a bit sad to see it go. I have been hoping that, in time, we are going to use the same sid as `sid` wherever components or objects are in the same session. For evals, I thought it's nice during debugging to recognize the name immediately, since it was just instance_id plus something. 😊  The event stream name is also set on the basis of the session id, so it was saved as instance_id_something, super easy to recognize.

But maybe it's not necessary since we now have both the contents of the stream and the completions available directly in EvalOutput. And we already rename the log using `instance_id`. So ok, if there's nothing else here, let's kill it in evals.

Reply:
Yeah i feel like the only place we will be needing runtime is actually where we want to pause/resume the session - but we are not doing that in eval, so it is probably easier to kill it. Also, we gave runtime a random ID as well, having an additional sid might cause more confusion.